shared/main.android.js

```diff
@@ -80,7 +80,7 @@ export default connect(
   }),
   dispatch => ({
     setRouteState: (path, partialState) => { dispatch(setRouteState(path, partialState)) },
-    navigateUp: () => { dispatch(navigateUp()) },
+    navigateUp: () => { dispatch(navigateUp(true)) },
```

Comment:
Oooh this is tricky. I'm not sure we should always persist state from back button presses -- in fact, pretty sure we shouldn't. I think we should move this into the route definition. Ticketing!